### PR TITLE
fix(domino): stop dropping the block mask off CUDA

### PR DESCRIPTION
**Pull request by khazic**
## What

DFlash and DSpark fall back to the dense equivalent of the flex block mask when the device is not CUDA, but the Domino backend still built the mask only under `device.type == "cuda"` and passed nothing otherwise. `DFlashAttention` then runs `scaled_dot_product_attention` with `attn_mask=None` and `is_causal=False`, so every draft query attends to the whole context, including the target tokens it is being trained to predict and every other draft block. On NPU or CPU the loss collapses by copying the leaked targets and the trained drafter is useless, with no error raised.

## Fix

Wire the shared `_create_dflash_dense_attention_mask` fallback into the Domino forward, exactly mirroring the DFlash backend's else branch, and pass `dense_attention_mask` through to the draft model (Domino inherits the DFlash forward, which already threads it into every layer).

## Tests

`test_domino_forward_passes_dense_mask_off_cuda` asserts that off CUDA the backend hands the draft model a boolean dense mask of shape `[bsz, 1, draft_len, seq_len + draft_len]` and no flex block mask. The existing CPU forward test now also exercises the masked path instead of unmasked attention.

### CODE REVIEW

File: tests/integration/test_domino_backend_contract.py [hunk 1]
```diff
@@ -98,6 +98,51 @@ def test_domino_forward_computes_top5_accuracy() -> None:
...
+        captured.update(kwargs)
+        return original_forward(*args, **kwargs)
+
+    monkeypatch.setattr(model.draft_model, "forward", spy)
+
+    bsz, seq_len = 2, 16
+    input_ids = torch.randint(0, config.vocab_size, (bsz, seq_len))
+    hidden_states_list = [torch.randn(bsz, seq_len, config.target_hidden_size) for _ in config.target_layer_ids]
+    loss_mask = torch.ones(bsz, seq_len, dtype=torch.long)
+    lm_head_weight = torch.randn(config.vocab_size, config.hidden_size)
```
[the remark this answers is not in the document]
Comment (khazic) [reply tests/integration/test_domino_backend_contract.py:134]: Not taking this one. The suggestion pins only the input tensors to CPU while the model itself is constructed without a device, so under a CUDA default device the params would land on CUDA and the inputs on CPU, which raises the device mismatch it claims to prevent rather than exercising the off-CUDA branch. The other tests in this file follow the same no-explicit-device convention and CI runs CPU-only, so the current form is both consistent and correct here.
